protocol/signalfx/log_hecv1_json.go

```diff
@@ -0,0 +1,66 @@
+package signalfx
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mailru/easyjson"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/logsink"
+	signalfxformat "github.com/signalfx/ingest-protocols/protocol/signalfx/format"
+)
+
+const (
+	// resources
+	host   = "host.hostname"
+	source = "service.name"
+	// attributes
+	sourceType = "source.type"
+	index      = "index"
+)
+
+// JSONDecoderHECV1 implements sink decoder for hecv1 splunk log events
+type JSONDecoderHECV1 struct {
+	Sink   logsink.Sink
+	Logger log.Logger
+}
+
+// Read implements interface method for decoding incoming json hecv1 format and convert it into golib/logsink.Log format
+func (decoder *JSONDecoderHECV1) Read(ctx context.Context, req *http.Request) (err error) {
+	var logHECV1List signalfxformat.JSONLogHECV1List
+	if err = easyjson.UnmarshalFromReader(req.Body, &logHECV1List); err == nil {
+
+		logs := make(logsink.Logs, len(logHECV1List))
+		for idx, lEntry := range logHECV1List {
+			e := &logsink.Log{
+				TimeStamp: uint64(*lEntry.Time),
+				Body:      lEntry.Event,
+				Resource: map[string]interface{}{
+					host:   lEntry.Host,
+					source: lEntry.Source,
```

Comment:
I am curious how this compiles. Shouldn't "host" and "source" be in quotes as strings?

Reply:
go magic :-)

Reply:
host & source are const see line 16

Reply:
I see, I missed that. It is good to have a const defined for this.

I am a bit confused by the names of consts. Why is "service.name" defined as a const named `source`? It looks like the name is given based on where the value is used not based on what it is, which I think makes it a bit difficult to understand.

Is "service.name" definition borrowed from OpenTelemetry semantic conventions for attributes? In that case perhaps it would be clearer to prefix them by `Attribute` and use name of the attribute in the name of the const, e.g. `AttributeServiceName` (see for example: https://github.com/open-telemetry/opentelemetry-collector/blob/8aa273184455591cad278c92c7cfcf75ad353d57/translator/conventions/opentelemetry.go#L20).

Up to you, I think this could improve the readability.

Reply:
made the changes @tigrannajaryan ... +100 on readability